src/app/WriteHandler.cpp

```diff
@@ -64,7 +65,8 @@ Status WriteHandler::OnWriteRequest(Messaging::ExchangeContext * apExchangeConte
     Status status = ProcessWriteRequest(std::move(aPayload), aIsTimedWrite);
 
     // Do not send response on Group Write
-    if (status == Status::Success && !apExchangeContext->IsGroupExchangeContext())
+    // Also, hold the response until finished processing the last chunk.
+    if (status == Status::Success && !apExchangeContext->IsGroupExchangeContext() && !mHasMoreChunks)
     {
         CHIP_ERROR err = SendWriteResponse();
```

Comment:
As per my comments above, write response is sent even when receiving chunks.

Reply:
Fixed